src/database/db.js

```diff
@@ -34,56 +36,120 @@ const Clip = mongoose.model('Clip', ClipSchema);
+        clipId,
+        { $push: { ratings: { rating, ratedBy } } },
+        { new: true }
+      );
+    } catch (error) {
+      console.error('Error adding rating:', error);
+      throw error;
+    }
   },
 
   // Get all clips
   getAllClips: async () => {
-    return await Clip.find();
+    try {
+      return await Clip.find();
+    } catch (error) {
+      console.error('Error getting all clips:', error);
+      throw error;
+    }
   },
 
   // Get a specific clip by ID
   getClipById: async (clipId) => {
-    return await Clip.findById(clipId);
+    try {
+      return await Clip.findById(clipId);
+    } catch (error) {
+      console.error('Error getting clip by ID:', error);
+      throw error;
+    }
   },
 
   // Get clips submitted by a specific user
   getClipsByUser: async (userId) => {
-    return await Clip.find({ submittedBy: userId });
+    try {
+      return await Clip.find({ submittedBy: userId });
+    } catch (error) {
+      console.error('Error getting clips by user:', error);
+      throw error;
+    }
   },
 
   // Get average rating for a clip
   getAverageRating: async (clipId) => {
-    const clip = await Clip.findById(clipId);
-    if (!clip || clip.ratings.length === 0) return 0;
-    const sum = clip.ratings.reduce((acc, r) => acc + r.rating, 0);
-    return sum / clip.ratings.length;
+    try {
+      const clip = await Clip.findById(clipId);
+      if (!clip || clip.ratings.length === 0) return 0;
+      const sum = clip.ratings.reduce((acc, r) => acc + r.rating, 0);
+      return sum / clip.ratings.length;
+    } catch (error) {
+      console.error('Error getting average rating:', error);
+      throw error;
+    }
   },
 
   // Get all ratings for a clip
   getRatingsForClip: async (clipId) => {
-    const clip = await Clip.findById(clipId);
-    return clip ? clip.ratings : [];
+    try {
+      const clip = await Clip.findById(clipId);
+      return clip ? clip.ratings : [];
+    } catch (error) {
+      console.error('Error getting ratings for clip:', error);
+      throw error;
+    }
   },
 
   // Get clips sorted by average rating
   getClipsSortedByRating: async () => {
-    return await Clip.aggregate([
-      { $unwind: "$ratings" },
-      { $group: { _id: "$_id", avgRating: { $avg: "$ratings.rating" }, clip: { $first: "$$ROOT" } } },
-      { $sort: { avgRating: -1 } }
-    ]);
+    try {
+      return await Clip.aggregate([
+        { $unwind: "$ratings" },
+        { $group: { _id: "$_id", avgRating: { $avg: "$ratings.rating" }, clip: { $first: "$$ROOT" } } },
+        { $sort: { avgRating: -1 } }
+      ]);
+    } catch (error) {
+      console.error('Error getting clips sorted by rating:', error);
+      throw error;
+    }
+  },
+
+  getClipByInteractionId: async (interactionId) => {
+    try {
+      return await Clip.findOne({ interactionId });
+    } catch (error) {
+      console.error('Error getting clip by interaction ID:', error);
+      throw error;
+    }
+  },
```

Comment:
This schema will attempt to find the clip by comparing your interactionId that you provided with the mongoDB unique generated id. Should define your query similar to how you did  with get clips by user